File: src/main/java/com/misiac/workoutjournal/entity/WorkoutExercise.java
package com.misiac.workoutjournal.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder

@Entity
@Table(name = "workout_exercises")
public class WorkoutExercise {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @JsonBackReference
    @ManyToOne(optional = false)
    @JoinColumn(name = "workout_id", nullable = false)
    private Workout parentWorkout;

    @ManyToOne(optional = false)
    @JoinColumn(name = "exercise_id", nullable = false)
    private Exercise exerciseType;

    @Column(name = "`load`", nullable = false)
    private Float load;

    @Column(name = "reps", nullable = false)
    private Integer reps;

    @Column(name = "set_number", nullable = false)
    private Integer setNumber;


}

File: wj-frontend/src/models/WorkoutExercise.ts
export class WorkoutExercise {
    counter: number;
    name: string;
    exerciseId: number;
    entry: WorkoutExerciseSet[];

    constructor(counter: number, name: string, exerciseId: number, entry: WorkoutExerciseSet[]) {
        this.counter = counter;
        this.name = name;
        this.exerciseId = exerciseId;
        this.entry = entry;
    }
}

export class WorkoutExerciseSet {
    id: number;
    load: number;
    reps: number;
    setNumber: number;

    constructor(id: number, load: number, reps: number, setNumber: number) {
        this.id = id;
        this.load = load;
        this.reps = reps;
        this.setNumber = setNumber;
    }
}

export default {WorkoutExercise, WorkoutExerciseSet};